src/BugsnagUnity/Payload/Metadata.cs

```diff
@@ -1,8 +1,26 @@
 ﻿using System.Collections.Generic;
+using BugsnagUnity;
 
 namespace BugsnagUnity.Payload
 {
   public class Metadata : Dictionary<string, object>, IFilterable
   {
+    private INativeClient NativeClient;
+
+    internal Metadata(INativeClient nativeClient) {
+      NativeClient = nativeClient;
+    }
+
+    public void Add(string section, object newValue) {
+      base.Add(section, newValue);
+      if (newValue is Dictionary<string, string> values) {
```

Comment:
I'm not sure I follow why the native client's metadata should be updated only if the parameter is a dictionary?

Reply:
The documented usage of this class is `Metadata.Add(string, dict<string, string>)` which cleanly maps to how metadata is structured. 

If the value is not a dictionary, then there is no representation for storing arbitrary objects and translating them over the native bridge.